File: lab2/python/serialize_json.py
"""
serialize json
"""
import json


class SerializeJson:
    @staticmethod
    def runToFile(deserialized_data, destination_file_locaiton):
        with open(destination_file_locaiton, 'w', encoding='utf-8') as f:
            json.dump(SerializeJson.get_data(deserialized_data), f, ensure_ascii=False)

    @staticmethod
    def run(deserialized_data):
        return json.dump(SerializeJson.get_data(deserialized_data))

    @staticmethod
    def get_data(deserialized_data):
        lst = []
        etykiety = ['Kod_TERYT', 'Województwo', 'Powiat', 'typ_JST', 'nazwa_urzędu_JST', 'miejscowość']
        for dep in deserialized_data.data:
            tmp_dep = {}
            for e in etykiety:
                tmp_dep[e] = dep.get(e)
            tmp_dep['telefon_z_numerem_kierunkowym'] = str(dep.get('telefon kierunkowy')) + " " + str(
                dep.get('telefon'))
            lst.append(tmp_dep)
        return {"departaments": lst}

File: lab2/csharp zadanie/csharp zadanie/SerializeJson.cs
using System.Text;
using System.Text.Json;

namespace csharp_zadanie;

public class SerializeJson
{
    public static void runToFile(List<Urzad> deserializedData, string path)
    {
        string serialized = JsonSerializer.Serialize(getData(deserializedData));
        File.WriteAllText(path, serialized, Encoding.UTF8);
    }
    
    public static string run(List<Urzad> deserializedData)
    {
        return JsonSerializer.Serialize(getData(deserializedData));
    }

    public static JsonDocument getData(IEnumerable<Urzad> data)
    {
        List<UrzadMod> lst = data.Select(dep => new UrzadMod(dep)).ToList();
        RejestrMod rejestrMod = new RejestrMod
        {
            Departments = lst
        };
        return JsonSerializer.SerializeToDocument(rejestrMod);
    }
}